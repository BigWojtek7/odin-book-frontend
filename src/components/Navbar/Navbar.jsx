import { Link, useLocation } from 'react-router-dom';
import styles from './Navbar.module.css';

import Icon from '@mdi/react';
import {
  mdiLogin,
  mdiAccountPlus,
  mdiHome,
  mdiAccount,
  mdiAccountGroup,
  mdiCogs,
} from '@mdi/js';

function Navbar() {
  const { pathname } = useLocation();
  console.log(pathname);
  return (
    <nav>
      <ul className={styles.navList}>
        <li className={pathname === '/' ? styles.active : undefined}>
          <Link to="/">
            <Icon path={mdiHome} size={1.4} />
            Home
          </Link>
        </li>
        <li className={pathname === '/sign-up' ? styles.active : undefined}>
          <Link to="sign-up">
            <Icon path={mdiAccountPlus} size={1.4} />
            SignUp
          </Link>
        </li>
        <li className={pathname === '/login' ? styles.active : undefined}>
          <Link to="login">
            <Icon path={mdiLogin} size={1.4} />
            Login
          </Link>
        </li>
        <li className={pathname === '/profile' ? styles.active : undefined}>
          <Link to="profile">
            <Icon path={mdiAccount} size={1.4} />
            Profile
          </Link>
        </li>
        <li>
          <Link to="/requests">
            <Icon path={mdiAccountGroup} size={1.4} />
            Requests
          </Link>
        </li>
        <li>
          <Link to="/settings">
            <Icon path={mdiCogs} size={1.4} />
            Settings
          </Link>
        </li>
      </ul>
    </nav>
  );
}
export default Navbar;
