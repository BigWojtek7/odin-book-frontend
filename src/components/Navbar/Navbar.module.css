nav {
  background-color: var(--nav-bg);
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  padding-block: 2rem;
  font-size: 1.25rem;
}

.navList {
  list-style: none;
  display: grid;
  gap: 1rem;
}

.navList a {
  padding-block: 1rem;
  padding-inline: 3rem 5rem;
  text-decoration: none;
  color: inherit;
  display: flex;
  align-items: center;
  gap: 5px;
}

.navList li:hover {
  background-color: var(--body-bg);
}

.navList li.active {
  display: block;
  background: var(--body-bg);
}

.navList li.active svg {
  color: var(--icon-clr);
}
