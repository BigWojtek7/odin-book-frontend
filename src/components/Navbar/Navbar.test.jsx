import { expect } from 'vitest';
import Navbar from './Navbar';

import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

describe('Testing Navbar Components', () => {
  it('The component UI should not change', () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
