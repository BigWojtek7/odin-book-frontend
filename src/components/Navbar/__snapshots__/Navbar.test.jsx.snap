// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing Navbar Components > The component UI should not change 1`] = `
<div>
  <div
    class="_navHeader_278c00"
  >
    <button
      class="_btnToggle_278c00"
    >
      <svg
        role="presentation"
        style="width: 2.0999999999999996rem; height: 2.0999999999999996rem;"
        viewBox="0 0 24 24"
      >
        <path
          d="M3,6H21V8H3V6M3,11H21V13H3V11M3,16H21V18H3V16Z"
          style="fill: currentColor;"
        />
      </svg>
    </button>
    <a
      class="_logo_278c00"
      href="/"
    >
      FriendsBook
    </a>
    <nav
      class="_nav_278c00 "
    >
      <ul
        class="_navList_278c00"
      >
        <li
          class="_active_278c00"
        >
          <a
            class="_navLink_278c00"
            href="/"
          >
            <svg
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M10,20V14H14V20H19V12H22L12,3L2,12H5V20H10Z"
                style="fill: currentColor;"
              />
            </svg>
            Home
          </a>
        </li>
        <li>
          <a
            class="_navLink_278c00"
            href="/profile"
          >
            <svg
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,4A4,4 0 0,1 16,8A4,4 0 0,1 12,12A4,4 0 0,1 8,8A4,4 0 0,1 12,4M12,14C16.42,14 20,15.79 20,18V20H4V18C4,15.79 7.58,14 12,14Z"
                style="fill: currentColor;"
              />
            </svg>
            Profile
          </a>
        </li>
        <li>
          <a
            class="_navLink_278c00"
            href="/requests"
          >
            <svg
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,5.5A3.5,3.5 0 0,1 15.5,9A3.5,3.5 0 0,1 12,12.5A3.5,3.5 0 0,1 8.5,9A3.5,3.5 0 0,1 12,5.5M5,8C5.56,8 6.08,8.15 6.53,8.42C6.38,9.85 6.8,11.27 7.66,12.38C7.16,13.34 6.16,14 5,14A3,3 0 0,1 2,11A3,3 0 0,1 5,8M19,8A3,3 0 0,1 22,11A3,3 0 0,1 19,14C17.84,14 16.84,13.34 16.34,12.38C17.2,11.27 17.62,9.85 17.47,8.42C17.92,8.15 18.44,8 19,8M5.5,18.25C5.5,16.18 8.41,14.5 12,14.5C15.59,14.5 18.5,16.18 18.5,18.25V20H5.5V18.25M0,20V18.5C0,17.11 1.89,15.94 4.45,15.6C3.86,16.28 3.5,17.22 3.5,18.25V20H0M24,20H20.5V18.25C20.5,17.22 20.14,16.28 19.55,15.6C22.11,15.94 24,17.11 24,18.5V20Z"
                style="fill: currentColor;"
              />
            </svg>
            Requests
          </a>
        </li>
        <li>
          <a
            class="_navLink_278c00"
            href="/settings"
          >
            <svg
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.9,18.45C17.25,18.45 18.35,17.35 18.35,16C18.35,14.65 17.25,13.55 15.9,13.55C14.54,13.55 13.45,14.65 13.45,16C13.45,17.35 14.54,18.45 15.9,18.45M21.1,16.68L22.58,17.84C22.71,17.95 22.75,18.13 22.66,18.29L21.26,20.71C21.17,20.86 21,20.92 20.83,20.86L19.09,20.16C18.73,20.44 18.33,20.67 17.91,20.85L17.64,22.7C17.62,22.87 17.47,23 17.3,23H14.5C14.32,23 14.18,22.87 14.15,22.7L13.89,20.85C13.46,20.67 13.07,20.44 12.71,20.16L10.96,20.86C10.81,20.92 10.62,20.86 10.54,20.71L9.14,18.29C9.05,18.13 9.09,17.95 9.22,17.84L10.7,16.68L10.65,16L10.7,15.31L9.22,14.16C9.09,14.05 9.05,13.86 9.14,13.71L10.54,11.29C10.62,11.13 10.81,11.07 10.96,11.13L12.71,11.84C13.07,11.56 13.46,11.32 13.89,11.15L14.15,9.29C14.18,9.13 14.32,9 14.5,9H17.3C17.47,9 17.62,9.13 17.64,9.29L17.91,11.15C18.33,11.32 18.73,11.56 19.09,11.84L20.83,11.13C21,11.07 21.17,11.13 21.26,11.29L22.66,13.71C22.75,13.86 22.71,14.05 22.58,14.16L21.1,15.31L21.15,16L21.1,16.68M6.69,8.07C7.56,8.07 8.26,7.37 8.26,6.5C8.26,5.63 7.56,4.92 6.69,4.92A1.58,1.58 0 0,0 5.11,6.5C5.11,7.37 5.82,8.07 6.69,8.07M10.03,6.94L11,7.68C11.07,7.75 11.09,7.87 11.03,7.97L10.13,9.53C10.08,9.63 9.96,9.67 9.86,9.63L8.74,9.18L8,9.62L7.81,10.81C7.79,10.92 7.7,11 7.59,11H5.79C5.67,11 5.58,10.92 5.56,10.81L5.4,9.62L4.64,9.18L3.5,9.63C3.41,9.67 3.3,9.63 3.24,9.53L2.34,7.97C2.28,7.87 2.31,7.75 2.39,7.68L3.34,6.94L3.31,6.5L3.34,6.06L2.39,5.32C2.31,5.25 2.28,5.13 2.34,5.03L3.24,3.47C3.3,3.37 3.41,3.33 3.5,3.37L4.63,3.82L5.4,3.38L5.56,2.19C5.58,2.08 5.67,2 5.79,2H7.59C7.7,2 7.79,2.08 7.81,2.19L8,3.38L8.74,3.82L9.86,3.37C9.96,3.33 10.08,3.37 10.13,3.47L11.03,5.03C11.09,5.13 11.07,5.25 11,5.32L10.03,6.06L10.06,6.5L10.03,6.94Z"
                style="fill: currentColor;"
              />
            </svg>
            Settings
          </a>
        </li>
        <li>
          <a
            class="_logout_278c00 _navLink_278c00"
            href="#"
          >
            <svg
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 7L15.59 8.41L18.17 11H8V13H18.17L15.59 15.58L17 17L22 12M4 5H12V3H4C2.9 3 2 3.9 2 5V19C2 20.1 2.9 21 4 21H12V19H4V5Z"
                style="fill: currentColor;"
              />
            </svg>
            Log Out
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
